include/exec/async_object.hpp

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2024 Kirk Shoop
+ * Copyright (c) 2023 NVIDIA Corporation
+ *
+ * Licensed under the Apache License Version 2.0 with LLVM Exceptions
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *   https://llvm.org/LICENSE.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "../stdexec/execution.hpp"
+#include "../stdexec/concepts.hpp"
+
+#include "__detail/__manual_lifetime.hpp"
+
+namespace exec {
+
+struct async_construct_t {
+  template<class _O, class _Stg, class... _An>
+  auto operator()(_O&& __o, _Stg& __stg, _An&&... __an) const 
+    noexcept(noexcept(((_O&&)__o).async_construct(__stg, ((_An&&)__an)...)))
+    -> std::enable_if_t<
+      !stdexec::same_as<decltype(((_O&&)__o).async_construct(__stg, ((_An&&)__an)...)), void>,
+      decltype(((_O&&)__o).async_construct(__stg, ((_An&&)__an)...))> {
```

Comment:
i think this condition would be better as a `static_assert` than as an `enable_if`. ditto for below. or (since these need to return senders, right?) maybe you can use `-> sender auto` as the return type.